execution.go

```diff
@@ -0,0 +1,41 @@
+package execution
+
+import (
+	"time"
+
+	"github.com/rollkit/rollkit/types"
+)
+
+// Execute defines a common interface for interacting with the execution client.
+type Execute interface {
```

Comment:
I have to disagree here. In Go generally interfaces should define behavior, what it does, not what it is. Otherwise you'd end up with what's called Interface Pollution.